pkg/metricsclient/faulty_sender.go

```diff
@@ -0,0 +1,80 @@
+package metricsclient
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strconv"
+	"time"
+
+	clientmodel "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/expfmt"
+)
+
+type FaultySender struct {
+	client      *http.Client
+	timeout     time.Duration
+	metricsName string
+}
+
+func NewFaultySender(client *http.Client, maxBytes int64, timeout time.Duration, metricsName string) *FaultySender {
+	return &FaultySender{
+		client:      client,
+		timeout:     timeout,
+		metricsName: metricsName,
+	}
+}
+
+func (c *FaultySender) Send(ctx context.Context, req *http.Request, families []*clientmodel.MetricFamily) error {
+	buf := &bytes.Buffer{}
+	if _, err := buf.Write([]byte{0xd, 0xe, 0xa, 0xd, 0xb, 0xe, 0xe, 0xf}); err != nil {
```

Comment:
Instead of making a buffer pointer and then writing the bytes to it, which can fail, why not create a new buffer directly from byte slice? https://golang.org/pkg/bytes/#NewBuffer